### PR TITLE
DOC-401: Add section to Drupal 9 guide to help users find their way

### PR DESCRIPTION
**Pull request by EdwardAngert**
Closes #DOC-401

## Summary

**[Migrate to Drupal 9 on Pantheon](https://pantheon.io/docs/guides/drupal-9-migration)** -  Directs users to the right part of the guide based on what their needs are.

## Effect

The following changes are already committed:

*
*

## Remaining Work

The following changes still need to be completed:

- [ ] fix the logic for fewest possible if/thens
- [ ] map out the use cases that we're going to document
  - [ ] include the ones we're not going to, and point them to the closest doc?
- [ ] bonus: adjust Previous/Next buttons on each page so that they navigate between logical steps instead of other use cases

--------------------------------------------------
## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6551](https://pr-6551--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `55.00000000000001` | `60` | `5` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-6551` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/28837/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/28837/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6551--pantheon-docs.my.pantheonfrontend.website/docs)

**Comment by EdwardAngert** [comment]
check out the [Quick Start guide](https://pantheon.io/docs/guides/quickstart/next-steps/) for how to set next and previous pages

**Comment by EdwardAngert** [comment]
[preview](https://pr-6551--pantheon-docs.my.pantheonfrontend.website/docs/guides/drupal-9-migration#migrate-or-upgrade-an-existing-site-to-drupal-9-with-integrated-composer-on-pantheon)

File: source/content/guides/drupal-9-migration/01-introduction.md [hunk 1]
```diff
@@ -18,4 +18,53 @@ Drupal 9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://s
...
+  - No Continuous Integration
+
+- Composer
+  - With [Integrated Composer](//guides/drupal-9-migration/upgrade-to-d9)
+  - Without Composer
+    - [On an Existing Site](/guides/drupal-9-migration/upgrade-to-d9)
+    - [New Site](/guides/drupal-9-migration/migrate-manual-d9)
+
+- Custom Upstream
+  - With Custom Upstream
```
Comment (michellecolon-pantheon) [inline source/content/guides/drupal-9-migration/01-introduction.md:46]: @EdwardAngert - Was not able to find a relevant link- could you give a look?

File: source/content/guides/drupal-9-migration/01-introduction.md [hunk 2]
```diff
@@ -18,4 +18,53 @@ Drupal 9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://s
...
+
+<Tab title="On Pantheon" id="on-pantheon" active={true}>
+
+Utilize the info on this tab if your site is currently on Pantheon.
+
+Here's how to migrate your site if you are using:
+
+- Build Tools (assumes Composer?)
+  - [Continuous Integration](/guides/drupal-9-migration/build-tools-to-d9-build-tools)
+  - No Continuous Integration
```
Comment (michellecolon-pantheon) [inline source/content/guides/drupal-9-migration/01-introduction.md:37]: @EdwardAngert - Was not able to find a relevant link- could you give a look?

Comment (EdwardAngert) [reply source/content/guides/drupal-9-migration/01-introduction.md:35]: just confirmed - it's scheduled in DOC-357 as work that hasn't been started yet

**Comment by EdwardAngert** [comment]
re-scoped and re-assigning to @wordsmither

**Pull request closed without merging**